runtime/cudaq/platform/default/rest_server/helpers/server_impl/RestServer.cpp

```diff
@@ -39,6 +39,10 @@ cudaq::RestServer::RestServer(int port, const std::string &name) {
   m_impl->app.port(port);
   m_impl->app.server_name(name);
   m_impl->app.loglevel(crow::LogLevel::Warning);
+  // Set a small stream threshold to enforce synchronous TCP buffer write.
+  // Asynchronous buffer write (`asio::async_write`) is susceptible to
+  // corruption if the app is shut down right after handling a request.
+  m_impl->app.stream_threshold(16 * 1024);
```

Comment:
Why not 0? If we set it to 16KB, wouldn't there still be an edge case if the `res.body.length()` is < 16KB?